src/main/java/ca/corefacility/bioinformatics/irida/model/sequenceFile/SequenceFile.java

```diff
@@ -307,4 +300,25 @@ public String getUploadSha256() {
 	public void setUploadSha256(String uploadSha256) {
 		this.uploadSha256 = uploadSha256;
 	}
+
+	@Override
+	public void setIridaFileStorageUtility(IridaFileStorageUtility iridaFileStorageUtility) {
+		this.iridaFileStorageUtility = iridaFileStorageUtility;
+	}
+
+	public InputStream getFileInputStream() {
+		return iridaFileStorageUtility.getFileInputStream(getFile());
+	}
+
+	public boolean isGzipped() throws Exception {
+		return iridaFileStorageUtility.isGzipped(getFile());
+	}
+
+	public File getTemporaryFile() {
```

Comment:
This is likely one that should stay with the `iridaFileStorageUtility`.  It's not really an action on the SequenceFile, it's a filesystem option.

Reply:
I can have it call the method directly in `iridaFileStorageUtility` but then we will need to add it to the fileprocessor methods. My reasoning behind it was so we don't need to add it to the files such as FastQcFileProcessor and can call the method getTemporaryFile on a `SequenceFile` object which would call it in the iridaFileStorageLocalUtility etc

Reply:
Yeah understood we would have to wire that utility in to those places, but it feels like a big mixing of concerns to have a model object creating temporary files.

That said, we should think whether the temp files should just be created with the regular `Files` object locally anyway.  For cloud systems we likely wouldn't want to create temporary files on the cloud storage service anyway would we?  Maybe a local temp file that's then transferred up to the object store is enough?

Reply:
We only use this method to get a temporary file from cloud storage to the local filesystem when we need to do something with the file. Like for example once a file is uploaded to the cloud and then the fastqcfileprocessor needs to run on the file. It will pull the file and run the processor

Reply:
Updated in 2f3e957f7a3a515362d5c8c107008236a1146e4d